common/utils/src/main/resources/error/error-conditions.json

```diff
@@ -3730,6 +3730,12 @@
     ],
     "sqlState" : "42K06"
   },
+  "STATE_STORE_KEY_SCHEMA_NOT_COMPATIBLE" : {
+    "message" : [
+      "Provided key schema does not match existing state key schema. Please check number and type of fields. Existing key_schema=<storedKeySchema> and new key_schema=<newKeySchema>."
```

Comment:
[nit] Do you think it would be useful to put existing and new key schema in separate lines for easier readability? Same comment for value schema error class below.

Reply:
Done